ivadomed/preprocessing.py

```diff
@@ -0,0 +1,46 @@
+import copy
+
```

Comment:
Another annoying comment but I wanted to ask:
What about: to keep everything as before in config files and loader and only modify `transforms.py` as follows:
- for all `__call__` of our preprocessing transforms (ie Resample and Crop): just before the return: add a metadata in this sample saying that "all good, preprocessing performed on this sample, no need to run it again"
- add a decorator, eg `@preprocessing` which check the metadata of the input sample --> if already done --> skip

Reply:
We'd still have perform the transforms in the load_filenames, because if we don't the images are not preprocessed -> if we keep the code as before we do the transform a first time, we change the metadata to indicate it has been done -> on the next getitem, the segpair retrived from index has not been cropped and resampled. I don't see how we can do this if we don't apply the specified transforms before calling getitem (in the load_filenames fonction), but I might not fully grasp what you're suggesting, so correct me if I'm wrong :)